File: MotivApp/app/src/main/java/com/motiv/piotr/dao/UsersListResponseDao.java
package com.motiv.piotr.dao;

import androidx.room.*;
import com.motiv.piotr.UsersListResponseWithReferences;

@Dao
public interface UsersListResponseDao {

    @Insert
    public long saveUsersListResponse(com.motiv.piotr.UsersListResponse usersListResponse);

    @Delete
    public int deleteUsersListResponse(com.motiv.piotr.UsersListResponse usersListResponse);

    @Query("SELECT * FROM usersListResponse")
    @Transaction
    public java.util.List<UsersListResponseWithReferences> getUsersListResponses();

    @Query("SELECT * FROM usersListResponse WHERE id = :id")
    @Transaction
    public UsersListResponseWithReferences getUsersListResponse(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void saveUsersListResponses(
            java.util.List<com.motiv.piotr.UsersListResponse> usersListResponse);
}

File: MotivApp/app/src/main/java/com/motiv/piotr/dao/UsersListResponseDao.kt
 
package com.motiv.piotr.dao
import androidx.room.*
import com.motiv.piotr.UsersListResponse
import com.motiv.piotr.UsersListResponseWithReferences
import kotlin.collections.List
@Dao
public interface UsersListResponseDao {

    @Insert
    public fun saveUsersListResponse(usersListResponse: com.motiv.piotr.UsersListResponse): Long@Delete
    public fun deleteUsersListResponse(usersListResponse: com.motiv.piotr.UsersListResponse): Int@Query("SELECT * FROM usersListResponse")
    @Transaction
    public fun getUsersListResponses(): List<UsersListResponseWithReferences>@Query("SELECT * FROM usersListResponse WHERE id = :id")
    @Transaction
    public fun getUsersListResponse(id: String): UsersListResponseWithReferences@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun saveUsersListResponses(usersListResponse: List<com.motiv.piotr.UsersListResponse>): Unit
}
